Schema update for IoT messages

varunpuranik opened the pull request:
- Updated the schema of the IotMessage so that the system properties
are structured into the source record being sent to the Kafka Connector.
- Changed exceptions thrown from IotHub to be ConnectExceptions to give
correct behavior in Kafka Connect
- Added additional tests for schema changes and also to show usage of
StartTime for the connector

varunpuranik committed:
Schema update for IoT messages

- Updated the schema of the IotMessage so that the system properties
are structured into the source record being sent to the Kafka Connector.
- Changed exceptions thrown from IotHub to be ConnectExceptions to give
correct behavior in Kafka Connect
- Added additional tests for schema changes and also to show usage of
StartTime for the connector

msftclas commented:
Hi __@varunpuranik__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Varun Puranik). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

dluc commented on build.sbt:
```diff
@@ -32,6 +32,8 @@ libraryDependencies ++= {
   )
 }
 
+assemblyJarName in assembly := "kafka-connect-iothub-assembly_2.11-0.5.jar"
```
tip: you could concatenate the `version` value, so that the jar filename will always match the current version

varunpuranik replied:
Good point, Done.

dluc commented on src/main/scala/com/microsoft/azure/iot/iothub2kafka/EventHubReceiver.scala:
```diff
@@ -7,7 +7,6 @@ import java.time.Instant
 import com.microsoft.azure.eventhubs.{EventHubClient, PartitionReceiver}
 
 import scala.collection.JavaConversions._
```
JavaConverters is the preferred approach over JavaConversions, so you can control when the transformations occur (using .asScala/.asJava when needed), with a possible performance gain + explicit behavior. Some info here for example: http://stackoverflow.com/questions/8301947/what-is-the-difference-between-javaconverters-and-javaconversions-in-scala
(same for other the files in the PR, in one case you are already using JavaConverters)

varunpuranik replied:
Makes sense, fixed.

dluc commented on src/main/scala/com/microsoft/azure/iot/iothub2kafka/IotMessageConverter.scala:
```diff
@@ -0,0 +1,85 @@
+
+  private lazy val propertiesMapSchema: Schema = SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.STRING_SCHEMA)
+
+  val offsetKey = "offset"
```
nit: could you move the constants at the top of the class?

varunpuranik replied:
The ide had formatted it like that. Fixed.

dluc commented on src/test/scala/com/microsoft/azure/iot/iothub2kafka/IotMessageConverterTest.scala:
```diff
@@ -0,0 +1,96 @@
+
+  private val random: Random = new Random
+
+  "IotMessage Converter"
```
something seems not right in these 2 lines

varunpuranik replied:
Emm, the string below was for the case description. Fixed the formatting.

dluc reviewed (approved):
Some minor changes nice-to-have, particularly removing JavaConversions in favor of JavaConverters

varunpuranik committed:
Updating README with schema update details

- Updated Readme with the schema for the message being written to Kafka
- Changed code to use JavaConverters instead of JavaConversions
- Other minor changes.
